src/wfs/src/main/java/org/geoserver/wfs/StoredQueryProvider.java

```diff
@@ -60,26 +84,45 @@ public String getLanguage() {
 
     /** Lists all the stored queries provided. */
     public List<StoredQuery> listStoredQueries() {
-        Parser p = new Parser(new WFSConfiguration());
-
-        List<StoredQuery> queries = new ArrayList();
+        final List<StoredQuery> queries = new ArrayList<>();
 
         // add the default as mandated by spec
         queries.add(StoredQuery.DEFAULT);
 
         // add user created ones
-        Resource dir = storedQueryDir();
-        for (Resource f : dir.list()) {
-            try {
-                queries.add(parseStoredQuery(f, p));
-            } catch (Exception e) {
-                LOGGER.log(Level.WARNING, "Error occured parsing stored query: " + f, e);
-            }
+        final Resource dir = storedQueryDir();
+        final List<StoredQuery> globalQueries = getStoredQueryByResource(dir);
+        final List<StoredQuery> localQueries = getLocalWorkspaceStoredQueries();
+        // add all global queries don't collide with local ones names
+        if (shouldProcessGlobalQueries()) {
+            globalQueries
+                    .stream()
+                    .filter(
+                            q ->
+                                    localQueries
```

Comment:
This is painful to see (the deep nesting and indenting), can get better looking code by storing this bit of code in its own method I believe.

Reply:
Indeed, refactoring applied.